k8s/k8s.go

```diff
@@ -3264,6 +3266,81 @@ func (k *k8sOps) DeleteMigrationSchedule(name string, namespace string) error {
 	})
 }
 
+func (k *k8sOps) ValidateMigrationSchedule(name string, namespace string, timeout, retryInterval time.Duration) error {
+	if err := k.initK8sClient(); err != nil {
+		return err
+	}
+	t := func() (interface{}, bool, error) {
+		resp, err := k.GetMigrationSchedule(name, namespace)
+		if err != nil {
+			return "", true, err
+		}
+
+		if len(resp.Status.Items) == 0 {
+			return "", true, &ErrFailedToValidateCustomSpec{
+				Name:  name,
+				Cause: fmt.Sprintf("0 migrations have yet run for the migration schedule"),
+				Type:  resp,
+			}
+		}
+
+		failedMigrations := make([]string, 0)
+		pendingMigrations := make([]string, 0)
+		for _, migrationStatuses := range resp.Status.Items {
```

Comment:
This can return false positives. If the caller wants to validate that a daily migration schedule has been triggered and there is only an interval schedule (just before the daily migration gets triggered) in successful state, it'll return a successful validation. The policy type to be validated also needs to be passed in, or return all the policy types that have a success and let the caller decide which one they want to check.

Reply:
Updated the method to return a map of successful migrations. I'm still calling the method Validate as it does validate that atleast one migration is successful.